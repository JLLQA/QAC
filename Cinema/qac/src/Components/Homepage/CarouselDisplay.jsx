import Carousel from 'react-bootstrap/Carousel';

const CarouselDisplay = () => {

    return (
        <Carousel>
            <Carousel.Item>
                <img
                    className="d-block"
                    src="https://ichef.bbci.co.uk/news/976/cpsprodpb/12B60/production/_109004667_02untitledgoosegamescreen3840x2160.png"
                    alt="First slide"
                />
                <Carousel.Caption>
                    <h3>First slide label</h3>
                    <p>Nulla vitae elit libero, a pharetra augue mollis interdum.</p>
                </Carousel.Caption>
            </Carousel.Item>
            <Carousel.Item>
                <img
                    className="d-block"
                    src="https://cdn.mos.cms.futurecdn.net/JycrJzD5tvbGHWgjtPrRZY-1200-80.jpg"
                    alt="Second slide"
                />
                <Carousel.Caption>
                    <h3>Second slide label</h3>
                    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
                </Carousel.Caption>
            </Carousel.Item>
            <Carousel.Item>
                <img
                    className="d-block"
                    src="https://ichef.bbci.co.uk/news/976/cpsprodpb/12B60/production/_109004667_02untitledgoosegamescreen3840x2160.png"
                    alt="Third slide"
                />
                <Carousel.Caption>
                    <h3>Third slide label</h3>
                    <p>Praesent commodo cursus magna, vel scelerisque nisl consectetur.</p>
                </Carousel.Caption>
            </Carousel.Item>
            <Carousel.Item>
                <img
                    className="d-block"
                    src="https://ichef.bbci.co.uk/news/976/cpsprodpb/12B60/production/_109004667_02untitledgoosegamescreen3840x2160.png"
                    alt="Fourth slide"
                />
                <Carousel.Caption>
                    <h3>Fourth slide label</h3>
                    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
                </Carousel.Caption>
            </Carousel.Item>
        </Carousel>
    )
}

export default CarouselDisplay;