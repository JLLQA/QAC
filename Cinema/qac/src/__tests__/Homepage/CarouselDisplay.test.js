import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import CarouselDisplay from '../../Components/Homepage/CarouselDisplay';
import renderer from 'react-test-renderer';

describe('test for carousel display component', () => {

test('renders learn react link', () => {
  render(<CarouselDisplay />);
  const linkElement = screen.getByText("Peter Rabbit 2");
  expect(linkElement).toBeInTheDocument();
});

test('It matches snapshot', () => {
    const tree = renderer.create(<CarouselDisplay/>).toJSON();
    expect(tree).toMatchSnapshot();
})

});