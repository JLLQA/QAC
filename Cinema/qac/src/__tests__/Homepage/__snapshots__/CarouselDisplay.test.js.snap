// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for carousel display component It matches snapshot 1`] = `
<div
  className="carousel slide"
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
>
  <ol
    className="carousel-indicators"
  >
    <li
      className="active"
      onClick={[Function]}
    />
    <li
      onClick={[Function]}
    />
    <li
      onClick={[Function]}
    />
    <li
      onClick={[Function]}
    />
  </ol>
  <div
    className="carousel-inner"
  >
    <div
      align="center"
      className="active carousel-item"
    >
      <img
        alt="First slide"
        className="d-block w-75 h-15"
        src="https://d1nslcd7m2225b.cloudfront.net/Pictures/780xany/3/5/4/1342354_pr2trlr00657_857435.jpg"
      />
      <div
        className="carousel-caption"
      >
        <h3
          id="CarouselName"
        >
          Peter Rabbit 2
        </h3>
      </div>
    </div>
    <div
      align="center"
      className="carousel-item"
    >
      <img
        alt="Second slide"
        className="d-block w-75 h-15"
        src="https://thenationroar.com/wp-content/uploads/2020/04/quiet-place-2.jpg"
      />
      <div
        className="carousel-caption"
      >
        <h3
          id="CarouselName"
        >
          The Quiet Place 2
        </h3>
      </div>
    </div>
    <div
      align="center"
      className="carousel-item"
    >
      <img
        alt="Third slide"
        className="d-block w-75 h-15"
        src="https://www.joblo.com/assets/images/joblo/news/2021/02/cruella-poster-disney.jpg"
      />
      <div
        className="carousel-caption"
      >
        <h3
          id="CarouselName"
        >
          Cruella
        </h3>
      </div>
    </div>
    <div
      align="center"
      className="carousel-item"
    >
      <img
        alt="Fourth slide"
        className="d-block w-75 h-15"
        src="https://movies-b26f.kxcdn.com/wp-content/uploads/2020/11/DSKY-Movie-Poster_FINAL-scaled-1-e1593800575920-770x470.jpg"
      />
      <div
        className="carousel-caption"
      >
        <h3
          id="CarouselName"
        >
          Demon Slayer
        </h3>
      </div>
    </div>
  </div>
  <a
    className="carousel-control-prev"
    href="#"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
  >
    <span
      aria-hidden="true"
      className="carousel-control-prev-icon"
    />
    <span
      className="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    className="carousel-control-next"
    href="#"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
  >
    <span
      aria-hidden="true"
      className="carousel-control-next-icon"
    />
    <span
      className="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;
